### PR TITLE
feat: Add more icons using glyphs from NerdFonts 3.2.0 release & more

### PR DESCRIPTION
**Pull request by hasecilu**
Recently a new version of Nerd Fonts came out and a lot of new icons were added (specially from [font-logos](https://github.com/lukas-w/font-logos/releases/tag/v1.1.0) project), so I'm looking to add a bunch of new icons for filetypes.

It seems that the `.fcbak` and `.fcstd1` are not recognized as TEMP extensions =(

```shell
# Command to recreate files
touch .aliases .bash_aliases .jwmrc .parentlock 2mm_tool.fctb 16x16matrix.kicad_dru 16x16matrix.kicad_pcb 16x16matrix.kicad_prl 16x16matrix.kicad_pro 16x16matrix.kicad_sch 123D.123dx Altium.pcbdoc Altium.prjpcb Altium.schdoc Aluminum-6061-T6.FCMat ass.ass AutoCAD.dwg AutoCAD.dxf block.scad brush.kpp bspwmrc CAD.ige CAD.iges CAD.igs CAD.ste CAD.step CAD.stp car.brep CATIA.catpart CATIA.catproduct CNC_machine.FCBak CNC_machine.FCStd CNC_machine.FCStd1 database.db3 database.s3db database.sl3 database.sqlite dl.crdownload drawing.kra drawing.krz drills_lib.fctl EAGLE.brd EAGLE.lbr EAGLE.sch en.pot es_MX.mo es_MX.po file.cache fp-info-cache fp-lib-table FreeCAD.conf Fusion_360.f3d Fusion_360.f3z hyprland.conf i3blocks.conf i3status.conf in3dca_storage.FCMacro Inventor.iam Inventor.ipt LED.kicad_sym LibreCAD_font.lff LibrePCB.lpp lock lxde-rc.xml lxqt.conf mail_header.msf My_page_layout.kicad_wks params.FCParam Parasolid.x_b Parasolid.x_t photo.xcf playlist.pls Plug_3.5mm_CUI_SP-3541.kicad_mod PrusaSlicer.ini PrusaSlicerGcodeViewer.ini Rhinoceros_3D.3dm robot.ino script.FCScript settings.dconf SketchUp.skp SolidEdge.psm SolidWorks.sldasm SolidWorks.sldprt song.aac song.lrc soundfont.sf2 soundfont.sfz sxhkdrc sym-lib-table system.jwmrc title.kdenlivetitle uBlock_Origin.xpi videotutorial.kdenlive Visual_Studio.vsix weston.ini xmobarrc xmobarrc.hs xmonad.hs 

```

image

### CODE REVIEW

**Comment by hasecilu** [comment]
I think those were all the extensions I could come up, I someone have a suggestion I could add it.

I was expecting for `FCStd1` and `FCBak` files to be faded like the `dl.crdownload` example but aren't...

**Review by cafkafk — CHANGES_REQUESTED** [review]
So, I think we'll wait for #565, but could I ask you tosquash your commits in the meanwhile?

**Comment by hasecilu** [comment]
Extra changes:
- I didn't find on the web `acc` as audio file, probably a typo because `aac` was on Audio category, removed `acc`.
- Add SQLite icon for extensions `.db3`, `.s3db`, `.sl3`.
- Added `scad`, `ino`, `fcmacro`, `fcscript` extensions to "source code files"
- Added `lrc` (song lyrics file) to SUBTITLES, I think fits better here than in TEXT.

Should playlists (`cue`, `m3u` `m3u8` `pls`) and `lrc` be added to Audio? These are more like audio-companion not audio themself.

The squash is pretty much done, only waiting for feedback.

File: src/output/icons.rs [hunk 1]
```diff
@@ -320,9 +357,16 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
...
+    "123dx"          => Icons::CAD,              // 󰻫
+    "3dm"            => Icons::CAD,              // 󰻫
+    "3g2"            => Icons::VIDEO,            // 
+    "3gp"            => Icons::VIDEO,            // 
+    "3gp2"           => Icons::VIDEO,            // 
+    "3gpp"           => Icons::VIDEO,            // 
+    "3gpp2"          => Icons::VIDEO,            // 
     "7z"             => Icons::COMPRESSED,       // 
     "a"              => Icons::OS_LINUX,         // 
-    "acc"            => Icons::AUDIO,            // 
```
Comment (CarterLi) [inline src/output/icons.rs:325]: I think this change is not related to nerdfonts 3.0 and should be moved to #565

Comment (hasecilu) [reply src/output/icons.rs:325]: Actually a lot of the changes here are not related to NF but I added here to make #565 ready to merge asap, but still there is no feedback there =(

File: src/output/icons.rs [hunk 2]
```diff
@@ -405,12 +457,15 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "dll"            => Icons::LIBRARY,          // 
     "dmg"            => Icons::DISK_IMAGE,       // 
     "doc"            => Icons::DOCUMENT,         // 
+    "docm"           => Icons::DOCUMENT,         // 
     "docx"           => Icons::DOCUMENT,         // 
-    "dot"            => '\u{f1049}',             // 󱁉
+    "dot"            => Icons::GRAPH,            // 󱁉
```
Comment (CarterLi) [inline src/output/icons.rs:462]: This is not related to nerdfonts 3.1.0 either

File: src/output/icons.rs [hunk 3]
```diff
@@ -658,14 +778,23 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
...
+    "sldasm"         => Icons::CAD,              // 󰻫
+    "sldprt"         => Icons::CAD,              // 󰻫
     "slim"           => Icons::LANG_RUBYRAILS,   // 
     "sln"            => '\u{e70c}',              // 
+    "slvs"           => Icons::CAD,              // 󰻫
     "so"             => Icons::OS_LINUX,         // 
     "sql"            => Icons::DATABASE,         // 
-    "sqlite3"        => '\u{e7c4}',              // 
+    "sqlite"         => Icons::SQLITE,           // 
+    "sqlite3"        => Icons::SQLITE,           // 
```
Comment (CarterLi) [inline src/output/icons.rs:791]: ditto

File: src/output/icons.rs (hunk 2 above)
Comment (hasecilu) [reply src/output/icons.rs:462]: That's why the PR says "** & more** at the end :wink:

File: src/output/icons.rs (hunk 1 above)
Comment (CarterLi) [reply src/output/icons.rs:325]: Changes require external upgrades are much harder to get merged. Bug fixes like this should be moved out

Comment (hasecilu) [reply src/output/icons.rs:326]: @cafkafk would you like this change to be moved to #565?

**Comment by cafkafk** [comment]
Is this still active?

**Comment by hasecilu** [comment]
> Is this still active?

Yes, it's been rebased again.

**Comment by hasecilu** [comment]
Now with the new feature to assign icons via a config file (kudos to devs) you can test on `main` the changes from this PR (+- some stuff).

<details>
  <summary><b>Click here to show/hide the config file</b></summary>

```yaml
filenames:
  .aliases: { icon: { glyph: 󱆃, style: { foreground: White } } }
  .bash_aliases: { icon: { glyph: 󱆃, style: { foreground: White } } }
  .envrc: { icon: { glyph: , style: { foreground: Blue } } }
  .git-blame-ignore-revs: { icon: { glyph: , style: { foreground: Red } } }
  .gtkrc-2.0: { icon: { glyph: , style: { foreground: White } } }
  .mailmap: { icon: { glyph: 󰊢, style: { foreground: Red } } }
  .parentlock: { icon: { glyph: , style: { foreground: White } } }
  .SRCINFO: { icon: { glyph: , style: { foreground: Blue } } }
  .Xauthority: { icon: { glyph: , style: { foreground: White } } }
  .xinitrc: { icon: { glyph: , style: { foreground: White } } }
  .Xresources: { icon: { glyph: , style: { foreground: White } } }
  .xsession: { icon: { glyph: , style: { foreground: White } } }
  bspwmrc: { icon: { glyph: , style: { foreground: White } } }
  cantorrc: { icon: { glyph: , style: { foreground: Blue } } }
  CODE_OF_CONDUCT: { icon: { glyph: , style: { foreground: Red } } }
  CODE_OF_CONDUCT.md: { icon: { glyph: , style: { foreground: Red } } }
  COMMIT_EDITMSG: { icon: { glyph: 󰊢, style: { foreground: Red } } }
  data: { icon: { glyph: , style: { foreground: White } } }
  favicon.ico: { icon: { glyph: , style: { foreground: Yellow } } }
  fonts.conf: { icon: { glyph: , style: { foreground: Green } } }
  FreeCAD.conf: { icon: { glyph: , style: { foreground: Red } } }
  fp-info-cache: { icon: { glyph: , style: { foreground: White } } }
  fp-lib-table: { icon: { glyph: , style: { foreground: White } } }
  gtkrc: { icon: { glyph: , style: { foreground: White } } }
  hypridle.conf: { icon: { glyph: , style: { foreground: Cyan } } }
  hyprland.conf: { icon: { glyph: , style: { foreground: Cyan } } }
  hyprlock.conf: { icon: { glyph: , style: { foreground: Cyan } } }
  i3blocks.conf: { icon: { glyph: , style: { foreground: White } } }
  i3status.conf: { icon: { glyph: , style: { foreground: White } } }
  input: { icon: { glyph: , style: { foreground: White } } }
  kalgebrarc: { icon: { glyph: , style: { foreground: Blue } } }
  kdeglobals: { icon: { glyph: , style: { foreground: Blue } } }
  kdenlive-layoutsrc: { icon: { glyph: , style: { foreground: Blue } } }
  kdenliverc: { icon: { glyph: , style: { foreground: Blue } } }
  kritadisplayrc: { icon: { glyph: , style: { foreground: Magenta } } }
  kritarc: { icon: { glyph: , style: { foreground: Magenta } } }
  lock: { icon: { glyph: , style: { foreground: White } } }
  LOCK: { icon: { glyph: , style: { foreground: White } } }
  log: { icon: { glyph: , style: { foreground: Yellow } } }
  LOG: { icon: { glyph: , style: { foreground: Yellow } } }
  lxde-rc.xml: { icon: { glyph: , style: { foreground: Black, background: White } } }
  lxqt.conf: { icon: { glyph: , style: { foreground: Blue } } }
  mpv.conf: { icon: { glyph: , style: { foreground: Magenta } } }
  PKGBUILD: { icon: { glyph: , style: { foreground: Blue } } }
  platformio.ini: { icon: { glyph: , style: { foreground: Red } } }
  PrusaSlicer.ini: { icon: { glyph: , style: { foreground: Red } } }
  PrusaSlicerGcodeViewer.ini: { icon: { glyph: , style: { foreground: Red } } }
  qt5ct.conf: { icon: { glyph: , style: { foreground: Green } } }
  qt6ct.conf: { icon: { glyph: , style: { foreground: Green } } }
  QtProject.conf: { icon: { glyph: , style: { foreground: Green } } }
  README.md: { icon: { glyph: 󰂺, style: { foreground: Yellow } } }
  SECURITY: { icon: { glyph: 󰒃, style: { foreground: Blue } } }
  SECURITY.md: { icon: { glyph: 󰒃, style: { foreground: Blue } } }
  sxhkdrc: { icon: { glyph: , style: { foreground: White } } }
  sym-lib-table: { icon: { glyph: , style: { foreground: White } } }
  tmux.conf: { icon: { glyph: , style: { foreground: Green } } }
  tmux.conf.local: { icon: { glyph: , style: { foreground: Green } } }
  vlcrc: { icon: { glyph: 󰕼, style: { foreground: Red } } }
  weston.ini: { icon: { glyph: , style: { foreground: Yellow } } }
  xmobarrc: { icon: { glyph: , style: { foreground: Magenta } } }
  xmobarrc.hs: { icon: { glyph: , style: { foreground: Magenta } } }
  xmonad.hs: { icon: { glyph: , style: { foreground: Magenta } } }
extensions:
  123dx: { icon: { glyph: 󰻫 } }
  3dm: { icon: { glyph: 󰻫 } }
  3g2: { icon: { glyph:  } }
  3gp: { icon: { glyph:  } }
  3gp2: { icon: { glyph:  } }
  3gpp: { icon: { glyph:  } }
  3gpp2: { icon: { glyph:  } }
  aac: { icon: { glyph:  } }
  app: { icon: { glyph: , style: { foreground: Green } } }
  applescript: { icon: { glyph: , style: { foreground: White } } }
  ass: { icon: { glyph: 󰨖, style: { foreground: Yellow } } }
  brd: { icon: { glyph: , style: { foreground: Green } } }
  brep: { icon: { glyph: 󰻫 } }
  c: { icon: { style: { foreground: Blue } } }
  cache: { icon: { glyph:  } }
  cast: { icon: { glyph:  } }
  catpart: { icon: { glyph: 󰻫 } }
  catproduct: { icon: { glyph: 󰻫 } }
  cow: { icon: { glyph: 󰆚 } }
  cpp: { icon: { style: { foreground: Blue } } }
  css: { icon: { style: { foreground: Blue } } }
  db3: { icon: { glyph:  } }
  dconf: { icon: { glyph:  } }
  docm: { icon: { glyph: , style: { foreground: Blue } } }
  dwg: { icon: { glyph: 󰻫 } }
  dxf: { icon: { glyph: 󰻫 } }
  editorconfig: { icon: { glyph: , style: { foreground: White } } }
  f3d: { icon: { glyph: 󰻫 } }
  f3z: { icon: { glyph: 󰻫 } }
  fcbak: { icon: { glyph: , style: { foreground: White } } }
  fcmacro: { icon: { glyph: , style: { foreground: Red } } }
  fcmat: { icon: { glyph: , style: { foreground: Red } } }
  fcparam: { icon: { glyph: , style: { foreground: Red } } }
  fcscript: { icon: { glyph: , style: { foreground: Red } } }
  fcstd: { icon: { glyph: , style: { foreground: Red } } }
  fcstd1: { icon: { glyph: , style: { foreground: Red } } }
  fctb: { icon: { glyph: , style: { foreground: Red } } }
  fctl: { icon: { glyph: , style: { foreground: Red } } }
  flc: { icon: { glyph:  } }
  flf: { icon: { glyph:  } }
  git: { icon: { glyph: 󰊢, style: { foreground: Red } } }
  gbl: { icon: { glyph: , style: { foreground: Green } } }
  gbo: { icon: { glyph: , style: { foreground: Green } } }
  gbp: { icon: { glyph: , style: { foreground: Green } } }
  gbr: { icon: { glyph: , style: { foreground: Green } } }
  gbs: { icon: { glyph: , style: { foreground: Green } } }
  gm1: { icon: { glyph: , style: { foreground: Green } } }
  gml: { icon: { glyph: , style: { foreground: Green } } }
  gresource: { icon: { glyph: , style: { foreground: White } } }
  gtl: { icon: { glyph: , style: { foreground: Green } } }
  gto: { icon: { glyph: , style: { foreground: Green } } }
  gtp: { icon: { glyph: , style: { foreground: Green } } }
  gts: { icon: { glyph: , style: { foreground: Green } } }
  h: { icon: { glyph: , style: { foreground: Magenta } } }
  h++: { icon: { glyph: , style: { foreground: Magenta } } }
  hc: { icon: { glyph: , style: { foreground: Yellow } } }
  hpp: { icon: { glyph: , style: { foreground: Magenta } } }
  html: { icon: { style: { foreground: Red } } }
  iam: { icon: { glyph: 󰻫 } }
  ifc: { icon: { glyph: 󰻫 } }
  ige: { icon: { glyph: 󰻫 } }
  iges: { icon: { glyph: 󰻫 } }
  igs: { icon: { glyph: 󰻫 } }
  info: { icon: { glyph:  } }
  ino: { icon: { glyph: , style: { foreground: Cyan } } }
  ipt: { icon: { glyph: 󰻫 } }
  json: { icon: { glyph: , style: { foreground: Yellow } } }
  json5: { icon: { glyph: , style: { foreground: Yellow } } }
  jsonc: { icon: { glyph: , style: { foreground: Yellow } } }
  jwmrc: { icon: { glyph: , style: { foreground: Blue } } }
  kdenlive: { icon: { glyph: , style: { foreground: Blue } } }
  kdenlivetitle: { icon: { glyph: , style: { foreground: Blue } } }
  kicad_dru: { icon: { glyph: , style: { foreground: White } } }
  kicad_mod: { icon: { glyph: , style: { foreground: White } } }
  kicad_pcb: { icon: { glyph: , style: { foreground: White } } }
  kicad_prl: { icon: { glyph: , style: { foreground: White } } }
  kicad_pro: { icon: { glyph: , style: { foreground: White } } }
  kicad_sch: { icon: { glyph: , style: { foreground: White } } }
  kicad_sym: { icon: { glyph: , style: { foreground: White } } }
  kicad_wks: { icon: { glyph: , style: { foreground: White } } }
  kpp: { icon: { glyph: , style: { foreground: Magenta } } }
  kra: { icon: { glyph: , style: { foreground: Magenta } } }
  krz: { icon: { glyph: , style: { foreground: Magenta } } }
  lbr: { icon: { glyph:  } }
  lck: { icon: { glyph: , style: { foreground: White } } }
  lff: { icon: { glyph:  } }
  lpp: { icon: { glyph: , style: { foreground: Green } } }
  log: { icon: { glyph: , style: { foreground: Yellow } } }
  lrc: { icon: { glyph: 󰨖, style: { foreground: Yellow } } }
  lua: { icon: { glyph: , style: { foreground: Blue } } }
  luac: { icon: { glyph: , style: { foreground: Blue } } }
  luau: { icon: { glyph: , style: { foreground: Blue } } }
  mo: { icon: { glyph: 󰗊 } }
  msf: { icon: { glyph: , style: { foreground: Blue } } }
  nfo: { icon: { glyph:  } }
  odb: { icon: { glyph:  } }
  pcbdoc: { icon: { glyph: , style: { foreground: Green } } }
  pls: { icon: { glyph: 󰨖, style: { foreground: Yellow } } }
  po: { icon: { glyph: 󰗊 } }
  pot: { icon: { glyph: 󰗊 } }
  prjpcb: { icon: { glyph: , style: { foreground: Green } } }
  psm: { icon: { glyph: 󰻫 } }
  py: { icon: { glyph: , style: { foreground: Yellow } } }
  pyw: { icon: { glyph: , style: { foreground: Yellow } } }
  qm: { icon: { glyph: 󰗊 } }
  qml: { icon: { glyph: , style: { foreground: Green } } }
  qrc: { icon: { glyph: , style: { foreground: Green } } }
  qss: { icon: { glyph: , style: { foreground: Green } } }
  s3db: { icon: { glyph:  } }
  scad: { icon: { glyph: , style: { foreground: Yellow } } }
  sch: { icon: { glyph: 󰭅 } }
  schdoc: { icon: { glyph: 󰭅 } }
  sf2: { icon: { glyph: 󰽰 } }
  sfz: { icon: { glyph: 󰽰 } }
  skp: { icon: { glyph: 󰻫 } }
  sl3: { icon: { glyph:  } }
  sldasm: { icon: { glyph: 󰻫 } }
  sldprt: { icon: { glyph: 󰻫 } }
  slvs: { icon: { glyph: 󰻫 } }
  sqlite: { icon: { glyph:  } }
  sqlite3: { icon: { glyph:  } }
  ste: { icon: { glyph: 󰻫 } }
  step: { icon: { glyph: 󰻫 } }
  stp: { icon: { glyph: 󰻫 } }
  sv: { icon: { glyph: 󰍛 } }
  svh: { icon: { glyph: 󰍛 } }
  tbc: { icon: { glyph: 󰛓 } }
  tcl: { icon: { glyph: 󰛓 } }
  tmux: { icon: { glyph: , style: { foreground: Green } } }
  toml: { icon: { glyph:  } }
  ui: { icon: { glyph: , style: { foreground: Blue } } }
  vhdl: { icon: { glyph: 󰍛 } }
  vsix: { icon: { glyph: 󰨞, style: { foreground: Magenta } } }
  wrl: { icon: { glyph: 󰆧 } }
  wrz: { icon: { glyph: 󰆧 } }
  xcf: { icon: { glyph: , style: { foreground: Magenta } } }
  xpi: { icon: { glyph:  } }
  x_b: { icon: { glyph: 󰻫 } }
  x_t: { icon: { glyph: 󰻫 } }
```
</details>

**Review by cafkafk — CHANGES_REQUESTED** [review]
there is a conflict that needs to be resolved before full review

**Comment by cafkafk** [comment]
I've tried to fix conflicts from merging the nerdfonts 3.3 pr

**Comment by hasecilu** [comment]
> I've tried to fix conflicts from merging the nerdfonts 3.3 pr

Thanks, looks good!

**Review by cafkafk — APPROVED** [review]
Okay, this is ready for merge, thanks for the contribution!
